scripts/refine-oauth-token-timeouts.sh

```diff
@@ -0,0 +1,747 @@
+function show_label() {
+  local message="${1?Please specify heading for the label!}"
+  local length="$(( ${#message} + 8 ))"
+  printf '%0.s#' $(seq 1 $length)
+  echo
+  echo -e "### $message ###"
+  printf '%0.s#' $(seq 1 $length)
+  echo
+}
+
+# Update token expiration or inactivity property on selected OAuth resource
+# (identified by resource type and name) by running 'oc patch' with provided
+# path and new value arguments on that resource
+function oc_patch_token_property() {
+  local resourceType="${1?Please specify OAuth or OAuthClient as a resource!}"
+  local resourceName="${2?Please specify the name of the resource!}"
+  local resourcePath="${3?Please specify the path to the target location!}"
+  local newValue="${4?Please specify the new value of the token property!}"
+
+  # Enumerate supported resource types
+  declare -ra allowedResourceTypes=( "oauth" "oauthclient" )
+
+  # OpenShift namespace to operate at
+  local namespace="openshift-config"
+
+  # Convert both resource type and name to lowercase
+  resourceType="$(echo "${resourceType}" | tr '[:upper:]' '[:lower:]')"
+  resourceName="$(echo "${resourceName}" | tr '[:upper:]' '[:lower:]')"
+
+  # Verify provided resource is either OAuth or OAuthClient
+  if [[ ! "${allowedResourceTypes[*]}" =~ ${resourceType} ]]
+  then
+    echo "Unsupported resource type: ${resourceType}!"
+    exit 1
+  fi
+
+  # Prepare the patch argument for 'oc patch' based on actual arguments
+  #
+  # NOTE: The following variable is intentionally passing 'newValue' argument
+  #       unquoted (without escaped double quotes), so the actual value is
+  #       passed as integer type to the underlying oc command. It will be
+  #       explicitly converted to string later, where necessary.
+  local patch="
+        [{
+          \"op\":    \"replace\",
+          \"path\":  \"${resourcePath}\",
+          \"value\": ${newValue}
+        }]"
+
+  # Issue the actual 'oc patch' command on the 'openshift-config' namespace
+  oc patch "${resourceType}" "${resourceName}" \
+    --type=json --namespace="${namespace}" \
+    --patch="${patch}"
+}
+
+# curl returns exit code of 0 (success) even if HTTP status code wasn't 200 OK
+# Thus parse the response and return the HTTP status code and response body
+# if the status code was within the 2xx (Successful) range of HTTP status
+# codes. Otherwise show an error message and exit with appropriate error
+function process_curl_response() {
+  local response="${1?Please specify the curl response to process!}"
+  # Trim trailing whitespace from status code
+  # shellcheck disable=SC2155
+  local responseStatusCode=$(sed -n '1s/[[:space:]]\+$//;1p' <<< "${response}")
+  # Collect response content from the first empty line (excluded) till the end
+  # of the response as a reponse body
+  # shellcheck disable=SC2016
+  # shellcheck disable=SC2155
+  local responseBody=$(sed -n '/^.$/,${n;p}' <<< "${response}")
```

Comment:
```
About to issue the following curl request:
curl --data client_id=admin-cli --data grant_type=password --data username=admin --data-urlencode password=mIQJ4m4DfaUSpg== --include --show-error --silent https://keycloak-edge-redhat-rhoam-rhsso.<REDACTED>/auth/realms/master/protocol/openid-connect/token
sed: 1: "/^.$/,${n;p}": extra characters at the end of p command
```
got an error when running this on mac os

Reply:
Right, will have a look / fix. Thanks, Pavel! (sorry, didn't try it on MacOSX).

Reply:
@psturc This should be fixed now. Tried the script multiple times on both Fedora 33 & FreeBSD 12.2 (just to ensure `sed` calls syntax is also compliant with *BSD variants of sed), and works for me now on both systems.

Thanks for catching this!